packages/p2p/src/components/modal-manager/modals/filter-modal/filter-modal-search.tsx

```diff
@@ -1,42 +1,46 @@
 import React from 'react';
-import { Field as FormField, Formik, Form } from 'formik';
+import { Field as FormField, Formik, Form, FormikValues } from 'formik';
 import debounce from 'lodash.debounce';
 import { Icon, Input } from '@deriv/components';
 import { isDesktop } from '@deriv/shared';
-import { observer } from 'mobx-react-lite';
+import { observer } from '@deriv/stores';
 import { localize } from 'Components/i18next';
 import { useStores } from 'Stores';
 import './filter-modal-search.scss';
 
 const FilterModalSearch = () => {
     const { buy_sell_store, my_profile_store } = useStores();
+    const { setIsFilterModalLoading } = buy_sell_store;
+    const { getPaymentMethodsList, setSearchResults, setSearchTerm } = my_profile_store;
 
     const returnedFunction = debounce(() => {
-        my_profile_store.getPaymentMethodsList();
+        getPaymentMethodsList();
     }, 1000);
 
-    const onSearch = ({ search }) => {
-        my_profile_store.setSearchTerm(search.trim());
+    const onSearch = ({ search }: { search: string }) => {
+        setSearchTerm(search.trim());
 
         if (!search.trim()) {
-            my_profile_store.setSearchResults([]);
+            setSearchResults([]);
             return;
         }
 
-        buy_sell_store.setIsFilterModalLoading(true);
+        setIsFilterModalLoading(true);
         returnedFunction();
     };
 
-    const onSearchClear = setFieldValue => {
+    const onSearchClear = (setFieldValue: (name: string, value: string) => void) => {
         setFieldValue('search', '');
-        my_profile_store.setSearchTerm('');
-        my_profile_store.setSearchResults([]);
+        setSearchTerm('');
+        setSearchResults([]);
     };
 
-    const onSearchKeyUp = submitForm => {
+    const onSearchKeyUp = (submitForm: VoidFunction) => {
+        let typing_timer: NodeJS.Timeout | undefined;
```

Comment:
not working

Reply:
How about this?

```
const onSearchKeyUp = submitForm => {
        const typing_timer = setTimeout(() => {
            submitForm();
        }, 1000);

        clearTimeout(typing_timer);
};
```

Reply:
wont the functionality change then if we change the order?

Reply:
it shouldn't right. Because we clean the timeout, so why we clean it before we even initialize it?

Reply:
there will be no previous value because the variable `typing_timer` scope is only for this function? 🤔

Reply:
@farhan-nurzi-deriv the way Nada wrote this is correct, if you check the docs and examples they all doing the same :thinking: 
Did you tried your suggestion? Did it work?

Reply:
My sugegestion doesnt works and initial implementation also doesnt works because it doesnt call cleartimeout. typing_timer is always undefined when we call clearTimeout , we need to move it out outside of the function 👁️ 👁️